### ISSUE TITLE
[gatsby-plugin-feed] 🚨BREAKING CHANGE ⚠️ builds now fail Config Validation with "title" is required

### ISSUE THREAD

**Issue by rdela** [gatsbyjs/gatsby#17100]
My builds on https://github.com/rdela/rdela.com started failing all of a sudden with a gatsby-plugin-feed error, `Config Validation with "title" is required`:

```
 ERROR #11321  PLUGIN

"gatsby-plugin-feed" threw an error while running the onPreBootstrap lifecycle:

[Config Validation]: "title" is required

  Error: [Config Validation]: "title" is required
  
  - gatsby-node.js:84 _callee$
    [rdela.com]/[gatsby-plugin-feed]/gatsby-node.js:84:19
```

Here is the PR from Renovate that introduced this bug/error: https://github.com/rdela/rdela.com/pull/186

```
gatsby-plugin-feed | dependencies | patch | 2.3.7 -> 2.3.9
```

By examining the link attached to 2.3.7 -> 2.3.9 in the PR description:
https://renovatebot.com/package-diff?name=gatsby-plugin-feed&from=2.3.7&to=2.3.9

I was able to track down https://github.com/gatsbyjs/gatsby/pull/16814 where @eyalroth writes 

> Note that the plugin will now require a title to be configured for each feed, as [per the documentation](https://www.gatsbyjs.org/packages/gatsby-plugin-feed/).

@sidharthachatterjee approved and merged the PR.

That bit in the readme was only added 5 months ago by @DSchau:
https://github.com/gatsbyjs/gatsby/commit/626cab461640bd6598806f3ec5b98e196f83dc9a

Dustin wrote at the time in the commit and PR description:

> I propose that we make this a minor version bump for this plugin, once merged.

@secretfader and @pieh approved the changes and Michal merged the PR:
https://github.com/gatsbyjs/gatsby/pull/12085

I think it goes without saying that this is a breaking change, and would have been whether or not the docs had said this all along, but they didn't. I cannot see how there is any way this should have happened under a [PATCH version](https://semver.org/).

Further, my RSS feed had a title the whole time anyway.

![Screen Shot 2019-08-26 at 160054](https://user-images.githubusercontent.com/74381/63729152-362b7280-c81b-11e9-8d8b-a034c8e6f859.png)

Here is my config at the time the breaking change was introduced:
https://github.com/rdela/rdela.com/blob/755f3c98ab727f647377941b84dcc35a5b020e29/gatsby-config.js#L110-L163

Please try to respect SemVer. I know how to track down errors like these, but I am sure there are people out there with broken builds scratching their heads who will not drop by to weigh in or raise their voice in the issues, even if they are able to fix their builds. Gatsby is difficult and complicated enough to use without surprise breakage like this.

I appreciate all of your hard work on Gatsby and that there were other bugs being fixed as these breaking changes were introduced. Yet none of those bugs effected me, but the breaking change in gatsby-plugin-feed 2.3.8 did. I am able to fix the build by adding the title field in my config, but I still do not understand why I should have to add redundant information that exists in `siteMetadata` in the same file to replicate the result I was already getting without it.

Please let me know if I have misunderstood something and if you spot any problems with my site, issues and PRs are very welcome, as always.

**Comment by eyalroth** [comment]
I created a PR to replace the requirement with a warning. Sorry for the inconvenience.

**Comment by eyalroth** [comment]
@rdela The issue description was on point, no need to apologize :)

To be honest I was too a little confused with the disparity between the documentation and the implementation of this plugin; the documentation declare that the title is required, but the implementation didn't enforce this requirement, didn't test it, and even made sure to handle the case a title is missing (along with tests for that case).

Yes, the feed basically borrows the site's URL. If I'm not mistaken -- and I may very well be because I haven't looked too deep into this -- it does so in quite an awkward way by fetching the title with the default query and then injecting it into the feed with the default serialization.

Imagine someone wanting to override either the query or the serialization, so now they would have to make sure to include this functionality as well. I find this cumbersome. There is also the issue of coupling the plugin with a certain conventional metadata structure, which might lead to problems in the future (see https://github.com/GatsbyCentral/gatsby-v2-starter-lumen/pull/9/ for example).

Furthermore, when having multiple feeds (for different categories, languages, etc), it would be discouraged to have the same title for both feeds.

All in all I do believe that moving forward the title should be set explicitly, but obviously this should be done gradually and shouldn't break any build at the moment.

**Comment by wardpeet** [comment]
We went forward with putting title as required because it was written down in the readme.  Seems like we were a bit too strongly about this.

**Comment by DavidWells** [comment]
This broke for me too.

Pegging version to older fixed for me https://github.com/DavidWells/davidwells.io/commit/b2121b2afba715f646d6733a4b4ea3385dc7528b

**Comment by JeffML** [comment]
I had some trouble tracking down where I was suppose to put the extra title info. In case anyone reading this is as dense as me, it is in **gatsby-config.js => plugins => resolve:gatsby-plugin-feed** at bottom of **feeds**, after output.xml:

```
 plugins: [
          ...,
    {
      resolve: 'gatsby-plugin-feed',...
      feeds: [{
          output: '/rss.xml',
          title: siteConfig.title  // fix for issue @https://github.com/gatsbyjs/gatsby/issues/17100
        }]
```

**Comment by sidharthachatterjee** [comment]
**_tl;dr_** Fix published in `gatsby-plugin-feed@2.3.11` thanks to @eyalroth 

Apologies for the trouble that this caused, everyone. `title` was documented as a required field (5 months ago as Ricky pointed out) so we (perhaps _naively_ in retrospect) agreed that it was reasonable to enforce this. 

> Please try to respect SemVer. I know how to track down errors like these, but I am sure there are people out there with broken builds scratching their heads who will not drop by to weigh in or raise their voice in the issues, even if they are able to fix their builds. Gatsby is difficult and complicated enough to use without surprise breakage like this.

@rdela Absolutely! We are typically very careful when it comes to semver and _do_ commit to respecting it 💯 We appreciate your detailed investigation and empathy towards users who might not be equipped to do the same 💜
